### PR TITLE
V2.x

### PR DESCRIPTION
**Pull request by subeeshcbabu**
- #30  - Uses `is-my-json-valid` validator by default, instead of `enjoi`. (Option to use `joi` schema - useful for hapi integration).
- #40 -  Dereference the api (using swagger parser) before building the route and validators 
- Uses `swagvali` module to build validators.
- es6

### CODE REVIEW

File: lib/builders/routes.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+//const Fs = require('fs');
```
Comment (shaunwarman) [inline lib/builders/routes.js]: lingering comment

File: lib/builders/routes.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    let fileResolver = Thing.isObject(handlers)
+        // Use the handlers object
+        ? Promise.resolve(handlers)
+        // Construct the handler object using `merge-object-files` for a valid dir path.
+        : (Thing.isString(handlers) ? merge(handlers, [ 'js' ]) : Promise.resolve(null));
+    let validatorResolver = buildValidator(apiResolver, { validated: true});
+
+    return Promise.all([ apiResolver, fileResolver, validatorResolver ]).then(resolved => {
+        let routes = [];
+        let [ api, files, validators ] = resolved;
```
Comment (shaunwarman) [inline lib/builders/routes.js]: Guessing order is maintained from `Promise.all` ?

File: lib/index.js [hunk 3]
```diff
@@ -1,71 +1,47 @@
...
+    //If basedir is falsy, Use the default basedir.
+    basedir = basedir || Path.dirname(Caller());
+    //If handlers is truthy, it should be a valid option.
+    if (handlers) {
+        Assert.ok(isString(handlers) || isObject(handlers), 'Expected handlers to be a string or an object.');
+        Assert.ok(!isString(handlers) || handlers.length, 'Expected handlers to be a non-empty string.');
+    } else {
+        //If handlers options is not set, use the default dir name `handlers` (only if the dir exists).
+        let defaultPath = Path.join(basedir, './handlers');
+        ////For a one time lookup and invocation during configuration,
```
Comment (shaunwarman) [inline lib/index.js]: extra `//`

File: lib/builders/routes.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
+//const Fs = require('fs');
+const { merge } = require('merge-object-files');
+const Thing = require('core-util-is');
+const Utils = require('../utils');
+const buildSecurity = require('./security');
+const buildValidator = require('swagvali');
```
Comment (shaunwarman) [inline lib/builders/routes.js]: Might be nice to order imports, but unnecessary 🤓

File: lib/builders/routes.js (hunk 2 above)
Comment (subeeshcbabu) [inline lib/builders/routes.js]: yes

**Comment by subeeshcbabu** [comment]
Can this be merged? Because the target is a new branch altogether - `v2.x`, can I proceed with the merge?

**Comment by subeeshcbabu** [comment]
merging this PR to `v2.x` branch. Will define the acceptance criterion and test cases for move to `v2.x`

**Pull request merged**